### PR TITLE
feat: Armor and accessory durability functions

### PR DESCRIPTION
**Pull request by ShadowCat117**
[An example function](https://github.com/Wynntils/Artemis/files/12336396/durability.txt), only shows when you have something equipped

### CODE REVIEW

File: common/src/main/java/com/wynntils/functions/InventoryFunctions.java [hunk 1]
```diff
@@ -19,6 +21,162 @@
 import net.minecraft.world.item.ItemStack;
 
 public class InventoryFunctions {
+    public static class AccessoryCurrentDurabilityFunction extends Function<Integer> {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/functions/InventoryFunctions.java]: You only need to have a single getDurability function; we have already helper generic functions for dealing with capped values (to get max and current value).

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/functions/InventoryFunctions.java]: Held item has a function for current, max and CappedValue durability, or are they just outdated now?

Comment (magicus) [reply common/src/main/java/com/wynntils/functions/InventoryFunctions.java]: We should not have individual functions returning the current/max part of capped values -- we should only return the capped value (and the function should not be named "capped"). There are probably a handful of functions that were introduced before we had full support for capped values in functions; ideally these should be removed but maybe we want to keep backwards compatibility.

We should not be adding new such functions.

Or did I misunderstand you?

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/functions/InventoryFunctions.java]: Ah ok, yeah that's what I meant so we're just keeping them so older functions don't break

Comment (magicus) [reply common/src/main/java/com/wynntils/functions/InventoryFunctions.java]: Yep. At some point I'd like to deprecate them and get rid of them. Maybe we need "upfixers" for functions too... or just warn folks that they will disappear in X days time, or whatever.

File: common/src/main/java/com/wynntils/functions/InventoryFunctions.java [hunk 2]
```diff
@@ -19,6 +21,50 @@
 import net.minecraft.world.item.ItemStack;
 
 public class InventoryFunctions {
+    public static class CappedAccessoryDurabilityFunction extends Function<CappedValue> {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/functions/InventoryFunctions.java]:
```suggestion
    public static class AccessoryDurabilityFunction extends Function<CappedValue> {
```

I don't think you need to include Capped in the name. Once again, it was used in some places when we introduced capped variables to distinguish them from the old version, but for new functions, this is just making the function name more verbose.

**Comment by ShadowCat117** [comment]
[Updated my example function, has weapon now too](https://github.com/Wynntils/Artemis/files/12340284/durability.txt)

**Review by magicus — APPROVED** [review]

**Comment by magicus** [comment]
Thanks! This looks very good now.

**Pull request merged**